assets/js/atomic/blocks/product-elements/rating/edit.tsx

```diff
@@ -3,22 +3,41 @@
  */
 import { __ } from '@wordpress/i18n';
 import { useBlockProps } from '@wordpress/block-editor';
+import type { ProductQueryContext as Context } from '@woocommerce/type-defs/contexts';
+import { useEffect } from 'react';
 
 /**
  * Internal dependencies
  */
 import Block from './block';
 import withProductSelector from '../shared/with-product-selector';
 import { BLOCK_TITLE, BLOCK_ICON } from './constants';
+import { Attributes } from './types';
 
-const Edit = ( { attributes } ) => {
+interface Props {
+	attributes: Attributes;
+	setAttributes: ( attributes: Record< string, unknown > ) => void;
+	context: Context;
+}
+
+const Edit = ( { attributes, setAttributes, context }: Props ): JSX.Element => {
 	const blockProps = useBlockProps( {
 		className: 'wp-block-woocommerce-product-rating',
 	} );
+	const blockAttrs = {
+		...attributes,
+		...context,
+	};
+	const isDescendentOfQueryLoop = Number.isFinite( context.queryId );
+
+	useEffect(
+		() => setAttributes( { isDescendentOfQueryLoop } ),
+		[ setAttributes, isDescendentOfQueryLoop ]
+	);
 
 	return (
 		<div { ...blockProps }>
-			<Block { ...attributes } />
+			<Block { ...blockAttrs } />
```

Comment:
I know this isn't specific from this PR, but I thought I would leave a comment here anyway. :upside_down_face: Would it be possible to pass `isDescendentOfQueryLoop` as a prop to `Block` so we don't need to save it in the block attributes?

Reply:
Thanks for leaving the comment! 
No, it isn't possible because this flag is [important for the save function](https://github.com/woocommerce/woocommerce-blocks/blob/8411854c52bbd5b0a97f19016c468cb5ad9532e8/assets/js/atomic/blocks/product-elements/save.js/#L6-L14)

Reading the attributes, when `isDescendentOfQueryLoop` is true we don't save anything, in this way on the PHP side, we know when to do server-side rendering or not https://github.com/woocommerce/woocommerce-blocks/blob/8411854c52bbd5b0a97f19016c468cb5ad9532e8/src/BlockTypes/ProductButton.php/#L77-L81

Reply:
Ah, right. In the PHP side, do you think we could invert the check, so instead of checking the content, we check for `$block->context['postId']` first? Similar to [what GB does](https://github.com/WordPress/gutenberg/blob/8f00bd0b129ad4afaef95b4f029787fcfc1c274c/packages/block-library/src/post-title/index.php#L18). So if there is a `postId`, we render the markup in PHP, if there isn't, we render the `content` as before.

Not sure if that would work (I just did some quick testing), but I feel it strange that we need to store `isDescendentOfQueryLoop` in the block attributes. :confused:

Reply:
Keep in mind that the GB block that you linked has only the "server side rendering version". Also, `context['postId']` is always available, even if the block isn't a child of the Product Query. The check that you linked is used for display or not the block if the post exists or not. We do the [same thing](https://github.com/woocommerce/woocommerce-blocks/blob/f8a6bab2ed31f1a0c4c0038c087dc20b33bd40a1/src/BlockTypes/ProductImage.php/#L196-L212).

We could use `queryId` instead of `postId` 🤔 

The problem is that we will save in `post_content` some HTML that we will not use... with the current approach we save nothing.

Reply:
Oooh, I see. Thanks for explaining. I think it's unfortunate that we need to save `isDescendentOfQueryLoop` as an attribute, but from what you say, it looks like there is no way we can get rid of it if we want the block to continue working in the context of the All Products block. :disappointed: 

I guess that if we migrate the All Products block to use the new Product Query block in the future, we will then be able to remove the `isDescendentOfQueryLoop` attribute.

Reply:
> Oooh, I see. Thanks for explaining.

You're welcome! 💪 

> I guess that if we migrate the All Products block to use the new Product Query block in the future, we will then be able to remove the isDescendentOfQueryLoop attribute.

Yeah! Exactly!